src/components/indicators/Speedometer.js

```diff
@@ -0,0 +1,41 @@
+import { useRef, useEffect } from 'react';
+import useSWR, { mutate } from 'swr'
+import { address, engineDataFetcher, putSpeed } from '../../services/SimulationService'
+
+const Speedometer = () => { 
+
+    const speedInputRef = useRef();
+
+    const { data, error, isValidating, mutate } = useSWR(address, engineDataFetcher);
+
+    let message = ""
+    if (isValidating)
+        message = `Loading vehicle speed from ${address} ...`
+    else if (error) 
+        message = `Failed vehicle speed from ${address}. ${error}.`
+    else
+        message = `Vehile speed is: ${data.vehicleSpeed}`
+
+    useEffect(() => {
+        if (data)
+            speedInputRef.current.value = data.vehicleSpeed;
+    }, [data])        
+
+    function updateSpeed() {
```

Comment:
try this , fa cateva incercari vezi cum se incarca foloseste si fetcheru cu eroare daca vrei. si dupaia doar schimba paginile intre ele sau da click pe orice fereastra, ideea e de simulare in timp real. tu actualizezi cache-ul doar dupa ce primesti raspunsul de la putSpeed. daca intra in eroare nu actualizeaza nimic. dupaia tu schimband ferestrele sau schimband focusul insine de pe acea fereastra datele se vor revalida.(:D doar da impresia de rapiditate)    
function updateSpeed() {
        var newSpeed = speedInputRef.current.value;
        const newData = {...data,vehicleSpeed:newSpeed}
        newData.setVehilceSpeed(newSpeed)
        putSpeed(address, newSpeed).then(()=>{mutate(newData,{revalidate:false})})
    }

Reply:
Ai pus revalidate:false si poate ca pentru un editor de text asta ar fi ok pentru ca poti oricand sa ii spui utilizatorului ca modificarile lui nu au putu fi salvate.

Dar gandeste ca ai o aplicatie bancara. 
Din frontend confirmi ca soldul a fost schimbat dar in backend ceva nu a mers bine.
Tu ii arati utilizatorului ca totul e in regula si utilizatorul e multumit pana data ce vine cand intra in aplicatie si isi da seama ca tranzactia nu a functionat.

In situatia asta as prefera sa arat un spinner pana imi vin datele citite din backend.

Reply:
Si mai e ceva.
Pe o aplicatie cu un backend real tu nu ai cum sa apelezi setVehilceSpeed.

Reply:
@ileuca vezi acum cu ii.
Am adaugat:
- front end update for data normally accessible to front end
- back end update in case of put error or success

Reply:
> Ai pus revalidate:false si poate ca pentru un editor de text asta ar fi ok pentru ca poti oricand sa ii spui utilizatorului ca modificarile lui nu au putu fi salvate.
> 
> Dar gandeste ca ai o aplicatie bancara. Din frontend confirmi ca soldul a fost schimbat dar in backend ceva nu a mers bine. Tu ii arati utilizatorului ca totul e in regula si utilizatorul e multumit pana data ce vine cand intra in aplicatie si isi da seama ca tranzactia nu a functionat.
> 
> In situatia asta as prefera sa arat un spinner pana imi vin datele citite din backend.

Asa e, ma gandeam ca o sa raspunzi cu asta  :D, nu voiam decat sa arat un usecase. pentru tabele si creare de inregistrari sau poate pentru pinuri si creare de pinuri da. merge de minune cu revalidate valse si actualizarea cache-ului.

In cazul in care backendul trebuie sa raspunda cu un calcul bazat pe ceea ce ii oferi tu nu ai cum. atunci raspunsul trebuie sa fie concret si da trebuie sa astepti raspunsul de pe backend.

Era doar un exemplu de usecase